cypress/integration/signup.spec.js

```diff
@@ -0,0 +1,48 @@
+/// <reference types="cypress" />
+
+describe("The Signup Page", () => {
+  function createRandomString(length) {
+    let result = "";
+    const characters =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    const charactersLength = characters.length;
+    for (let i = 0; i < length; i++) {
+      result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+  }
+  beforeEach(() => {
+    cy.visit("http://localhost:3000/signup");
+  });
+  it("sign ups user and logs in when form filled in correctly", () => {
+    cy.get("#username").type(createRandomString(5));
+    cy.get("#password").type("blahdeblah");
+    cy.get("#email").type(`${createRandomString(5)}@${createRandomString(5)}`);
```

Comment:
Why are you testing with random strings? Is it to avoid collisions with already existing username? You should be able to clear the user table before your tests if that's the case.
It's always better to tests with fixed input, otherwise your tests can be flaky (the same test passes sometimes and fils sometimes) which is really bad for tests.

Reply:
Yes, to avoid collisions using existing username. Currently, my feature tests are using the 'dev' database rather than 'test' database and so there isn't the 'tear down' in place that removes all of the test data. I did some research on this before but couldn't quite work out how to do feature tests using 'test' database. I will have another look and see if I can come up with a solution and then change the random strings to fixed strings